util/data_set_helpers.py

```diff
@@ -10,184 +9,131 @@
+        return source, source_lengths, sparse_labels
 
     def start_queue_threads(self, session, coord):
         batch_threads = []
-        for s in self._sets:
-            batch_threads += s.start_queue_threads(session, coord)
+        for set_queue in self._set_queues:
+            batch_threads += set_queue.start_queue_threads(session, coord)
         return batch_threads
 
-    def close_queue(self, session):
-        for s in self._sets:
-            s.close_queue(session)
-        session.run(self._close_op, feed_dict={ self._queue_selector: 0 })
-
-    def next_batch(self):
-        source, source_lengths, target, target_lengths = self._queue.dequeue_many(self._data_set.batch_size)
-        sparse_labels = ctc_label_dense_to_sparse(target, target_lengths, self._data_set.batch_size)
-        return source, source_lengths, sparse_labels
-
-def read_data_sets(train_csvs, dev_csvs, test_csvs,
-                   train_batch_size, dev_batch_size, test_batch_size,
-                   numcep, numcontext, thread_count=8,
-                   stride=1, offset=0, next_index=lambda s, i: i + 1,
-                   limit_dev=0, limit_test=0, limit_train=0):
-    # Read the processed set files from disk
-    def read_csvs(csvs):
-        files = None
-        for csv in csvs:
-            file = pandas.read_csv(csv)
-            if files is None:
-                files = file
-            else:
-                files = files.append(file)
-        return files
-
-    train_files = read_csvs(train_csvs)
-    dev_files = read_csvs(dev_csvs)
-    test_files = read_csvs(test_csvs)
-
-    # Create train DataSet from all the train archives
-    train = _read_data_set(train_files, thread_count, train_batch_size, numcep, numcontext, stride=stride, offset=offset, next_index=lambda i: next_index('train', i), limit=limit_train)
-
-    # Create dev DataSet from all the dev archives
-    dev = _read_data_set(dev_files, thread_count, dev_batch_size, numcep, numcontext, stride=stride, offset=offset, next_index=lambda i: next_index('dev', i), limit=limit_dev)
-
-    # Create test DataSet from all the test archives
-    test = _read_data_set(test_files, thread_count, test_batch_size, numcep, numcontext, stride=stride, offset=offset, next_index=lambda i: next_index('test', i), limit=limit_test)
-
-    # Return DataSets
-    return DataSets(train, dev, test)
-
-def _read_data_set(filelist, thread_count, batch_size, numcep, numcontext, stride=1, offset=0, next_index=lambda i: i + 1, limit=0):
-    # Optionally apply dataset size limits
-    if limit > 0:
-        filelist = filelist.iloc[:limit]
-
-    filelist = filelist[offset::stride]
-
-    # Return DataSet
-    return DataSet(filelist, thread_count, batch_size, numcep, numcontext, next_index=next_index)
+    def close_queues(self, session):
+        for set_queue in self._set_queues:
+            set_queue.close_queue(session)
+        session.run(self._close_op, feed_dict={ self.data_sets.ph_queue_selector: 0 })
```

Comment:
This closes the combining list-queue - it just requires it's queue-selection placeholder (with a dummy value) passed-in to accept the close call.